graph/src/components/store.rs

```diff
@@ -57,6 +59,40 @@ pub struct StoreRange {
     pub skip: usize,
 }
 
+#[derive(Clone, Debug, PartialEq)]
+pub enum ValueType {
+    Boolean,
+    ID,
+    Int,
+    Float,
+    String,
+    Bytes,
+    BigInt,
+}
+
+impl From<s::Type> for ValueType {
+    fn from(schema_type: s::Type) -> ValueType {
+        match schema_type {
+            s::Type::NonNullType(ref inner) => match inner.deref() {
+                s::Type::NamedType(ref name) if name == "Boolean" => ValueType::Boolean,
+                s::Type::NamedType(ref name) if name == "BigInt" => ValueType::BigInt,
+                s::Type::NamedType(ref name) if name == "Bytes" => ValueType::Bytes,
+                s::Type::NamedType(ref name) if name == "Float" => ValueType::Float,
+                s::Type::NamedType(ref name) if name == "ID" => ValueType::ID,
+                s::Type::NamedType(ref name) if name == "String" => ValueType::String,
+                _ => panic!("Found unexpected NamedType variant when converting to ValueType"),
+            },
+            s::Type::NamedType(ref name) if name == "Boolean" => ValueType::Boolean,
+            s::Type::NamedType(ref name) if name == "BigInt" => ValueType::BigInt,
+            s::Type::NamedType(ref name) if name == "Bytes" => ValueType::Bytes,
+            s::Type::NamedType(ref name) if name == "Float" => ValueType::Float,
+            s::Type::NamedType(ref name) if name == "ID" => ValueType::ID,
+            s::Type::NamedType(ref name) if name == "String" => ValueType::String,
+            _ => panic!("Found unexpected NamedType variant when converting to ValueType"),
+        }
```

Comment:
It looks like `ValueType` could use a [FromStr](https://doc.rust-lang.org/std/str/trait.FromStr.html) impl.

Reply:
That would indeed cleanup the From<s::Type> impl.  I'll add it in.